### PR TITLE
Experimental Transfer table and Management API

### LINKED ISSUE #2024: Transfers Management API

### PR DESCRIPTION
**Pull request by ysv**
* Add Transfers model
* Add Transfers /new & specs
* Add Entities::Transfer & specs
* Replace Operations::Chart with Operations::Account
* Implement chart of accounts seed file & rake task
* Require account code on Operation create
* Refactor Operations /new endpoints
* Management::Helpers #create_operation!
* Refactor Operations #credit! #debit!
* Remove Operations::Liability #credit! #debit!
* Allow null for Operations::Liability member_id
* Add Revenue member_id (revenues source)
* Add Operations::Account & Transfer factories
* Update Revenues record specs
* Create list of default accounts on each spec
    
closes #2024

### CODE REVIEW

File: app/models/operations/liability.rb [hunk 1]
```diff
@@ -5,69 +5,25 @@ module Operations
   class Liability < Operation
     belongs_to :member
 
-    class << self
-      def credit!(amount:, kind: :main, reference: nil,
-                  member_id: nil, currency: nil)
-        return if amount.zero?
+    validates :member_id, presence: {
+      if: ->(liability) { liability.account.scope.to_s == 'member' }
```
Comment (ysv) [inline app/models/operations/liability.rb]: Remove to_s

File: app/models/operation.rb [hunk 2]
```diff
@@ -6,52 +6,71 @@
...
+    end
+  end
 
   self.abstract_class = true
 
-  MEMBER_TYPES = %i[liability].freeze
-  PLATFORM_TYPES = %i[asset expense revenue].freeze
-  TYPES = (MEMBER_TYPES + PLATFORM_TYPES).freeze
+  # TODO: Remove TYPES.
+  MEMBER_TYPES = Operations::Account::MEMBER_TYPES
```
Comment (ysv) [inline app/models/operation.rb]: Remove this and usage

File: app/models/operation.rb [hunk 3]
```diff
@@ -70,4 +89,9 @@ def balance(currency: nil, created_at_from: nil, created_at_to: nil)
       end
     end
   end
+
```
Comment (ysv) [inline app/models/operation.rb]: Remove

File: spec/api/v2/management/operations_spec.rb [hunk 4]
```diff
@@ -159,9 +160,9 @@ def request(op_type)
             expect(JSON.parse(response.body)['currency']).to eq currency.code.to_s
             expect(JSON.parse(response.body)['credit'].to_d).to eq amount
             expect(JSON.parse(response.body)['code']).to \
-              eq Operations::Chart.code_for(type: op_type,
+              eq Operations::Account.find_by(type: op_type,
```
Comment (ysv) [inline spec/api/v2/management/operations_spec.rb]: use find_by! everywhere !!!!!!!!!!!!!!!!!!!

File: spec/api/v2/management/operations_spec.rb [hunk 5]
```diff
@@ -246,14 +257,25 @@ def request(op_type)
...
 
         context 'debit' do
           let(:amount) { 0.1545 }
           before do
             # Create credit operation to avoid negative balance.
-            create(op_type, credit: amount, member: member, currency: currency)
+            create(op_type, :with_member, credit: amount,
+                   member: member, currency: currency)
             data[:debit] = amount
+            # binding.pry
```
Comment (ysv) [inline spec/api/v2/management/operations_spec.rb]: remove

**Review by ysv — COMMENTED** [review]
remove peatio_member_api

**Comment by ianeinser** [comment]
Hi @ysv @mod ,

there are some new parameters in transfer such as unique transfer key, source account code, destination account code. How are they generated? Who generates them? I appreciate any clarification about that. 

regards,
ian

**Pull request merged**